### PR TITLE
Populate meta.pkgConfigModules for more packages

### PR DESCRIPTION
**Pull request by maralorn**
- libdatrie: add meta.pkgConfigModules annotation
- libepoxy: add meta.pkgConfigModules annotation
- libpsl: add meta.pkgConfigModules annotation
- libsoup: add meta.pkgConfigModules annotation
- libxkbcommon: add meta.pkgConfigModules annotation
- sqlite: add meta.pkgConfigModules annotation
- libsysprof-capture: add meta.pkgConfigModules annotation
- xorg.libxcb: add meta.pkgConfigModules annotation
- xorg.libXdmcp: add meta.pkgConfigModules annotation
- xorg.libXtst: add meta.pkgConfigModules annotation

A continuation of #247907.

/cc @sternenseemann 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libdatrie/default.nix [hunk 1]
```diff
@@ -42,5 +42,6 @@ stdenv.mkDerivation rec {
     license = licenses.lgpl21Plus;
     platforms = platforms.unix;
     maintainers = with maintainers; [ ];
+    pkgConfigModules = [ "datrie-2.0" ];
```
Comment (sternenseemann) [inline pkgs/development/libraries/libdatrie/default.nix:45]:
```suggestion
    pkgConfigModules = [ "datrie-0.2" ];
```

File: pkgs/development/libraries/libxkbcommon/default.nix [hunk 2]
```diff
@@ -63,5 +63,9 @@ stdenv.mkDerivation rec {
     maintainers = with maintainers; [ primeos ttuegel ];
     mainProgram = "xkbcli";
     platforms = with platforms; unix;
+    pkgConfigModules = [
+      "xkbcommon"
```
Comment (sternenseemann) [inline pkgs/development/libraries/libxkbcommon/default.nix:67]:
```suggestion
      "xkbcommon"
      "xkbcommon-x11
```

**Review by sternenseemann — APPROVED** [review]

File: pkgs/development/libraries/libdatrie/default.nix (hunk 1 above)
Comment (maralorn) [reply pkgs/development/libraries/libdatrie/default.nix:45]: Slightly flabbergasted how this annotation fixed my build although it was wrong.

Comment (maralorn) [reply pkgs/development/libraries/libdatrie/default.nix:45]: Its actually not the first time I noticed that. For libxcb I accidentally put "xcb.pc" in the pkgConfigModules list and it still fixed my dependency.

Comment (maralorn) [reply pkgs/development/libraries/libdatrie/default.nix:45]: Oh, we are only checking whether the attribute exists …

**Pull request merged**